jf_agent/main.py

```diff
@@ -561,19 +550,46 @@ def _s3_cmd(cmd):
 
     print('Sending data to Jellyfish... ')
 
-    output_basedir, output_dir_timestamp = os.path.split(outdir)
-    s3_uri_prefix_with_timestamp = os.path.join(s3_uri_prefix, output_dir_timestamp)
-    done_file_path = f'{os.path.join(outdir, ".done")}'
+    signed_urls = get_signed_url(os.listdir(config.outdir))
+
+    threads = []
+    thread_exceptions = [None] * (len(signed_urls) + 1)
+    for index, file_info in list(enumerate(signed_urls)):
+        # file_info shape {filename: (path_to_obj, signed_url object)}
+        filename = list(file_info.keys())[0]
+        touple = file_info[filename]
+        path_to_obj = touple[0]
+        signed_url = touple[1]
+
+        s3_upload_process = threading.Thread(
+            target=upload_file,
+            args=[index, filename, path_to_obj, signed_url],
+        )
+        s3_upload_process.start()
+        threads.append(s3_upload_process)
+    
+    print(f'Joining {len(threads)} threads')
+
+    for process in threads:
+        process.join()
+
+    # if all threads failed
+    if any(thread_exceptions):
+        status = {'type': 'S3 Upload', 'status': 'failed'}
+    else:
```

Comment:
Perhaps raise the first one, if there's at least one exception?   And/or log them here?  I think as-is, if a thread fails, we'll never find out the details of what happened.